doc/how-to-analyze-haskell-code-performance.rst

```diff
@@ -0,0 +1,124 @@
+How to analyze Haskell performance
+==================================
+
+When a Haskell application is slow or uses too much memory,
+Cabal and the
+`Haskell GHC compiler <https://downloads.haskell.org/ghc/latest/docs/users_guide/>`__
+can help you understand why.
+
+The main steps are to
+
+1. let GHC insert performance measuring code into your application,
+2. run the application to collect a performance report and
+3. visualize that report.
+
+The process of inserting performance measuring code and collecting performance information
+is called "profiling". Cabal makes this easy to do.
+
+Profiling CPU performance
+-------------------------
+
+First, build your application, e.g. ``my-app``, with profiling enabled:
+
+.. code-block:: console
+
+    $ cabal build --enable-profiling --profiling-detail=late my-app
+
+Under the hood, this will pass the corresponding
+`profiling compiler options <https://downloads.haskell.org/ghc/latest/docs/users_guide/profiling.html#compiler-options-for-profiling>`__
+to GHC. The additional Cabal flag ``--profiling-detail=late`` instructs GHC to use
```

Comment:
There should be a mention of compatible GHC versions. Late appeared only in 9.10, I think. And, for instance, ghcup still recommends 9.4.

Reply:
It was introduced in 9.4 (https://downloads.haskell.org/ghc/9.4.1/docs/users_guide/9.4.1-notes.html#compiler).

Reply:
Oh, I mixed it up with something else then. Thanks for catching it!

I retract the comment

Reply:
I’m fine with mentioning the version but I think it’s fine either way because 9.4 is pretty old by now, I feel like